src/metabase/api/api_key.clj

```diff
@@ -129,4 +129,19 @@
   (let [api-keys (t2/hydrate (t2/select :model/ApiKey) :group_name :updated_by)]
     (map present-api-key api-keys)))
 
+(api/defendpoint DELETE "/:id"
+  "Delete an ApiKey"
+  [id]
+  {id ms/PositiveInt}
+  (api/check-superuser)
+  (t2/with-transaction [_tx]
+    (let [api-key (-> (t2/select-one :model/ApiKey id)
+                      (t2/hydrate :group_name))]
+      (t2/delete! :model/ApiKey id)
+      (t2/update! :model/User (:user_id api-key) {:is_active false})
+      (events/publish-event! :event/api-key-delete
+                             {:object api-key
+                              :user-id api/*current-user-id*})
```

Comment:
Here's another case where I don't think the `publish-event!` call should be in the transaction, since we don't know what downstream DB calls are generated

Reply:
Oops, thanks for noticing this - sorry about that!